lms/djangoapps/program_enrollments/signals.py

```diff
@@ -0,0 +1,47 @@
+"""
+Signal handlers for program enrollments
+"""
+import logging
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from social_django.models import UserSocialAuth
+from student.models import CourseEnrollmentException
+from third_party_auth.models import SAMLProviderConfig
+
+from lms.djangoapps.program_enrollments.models import ProgramEnrollment
+
+logger = logging.getLogger(__name__)
+
+
+@receiver(post_save, sender=UserSocialAuth)
+def populate_waiting_enrollments(sender, instance, **kwargs):  # pylint: disable=unused-argument
```

Comment:
#NIT I think the name maybe better as `activate_enrollments` or `martriculate_learner`